addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,649 @@
+                "u_move_line_key_format": "{package_id.name}",
+            }
+        )
+
+        super(TestAssignSplitting, self).setUp()
+
+    def test_reserve_1_pallet_diff_prods_splits_1_pick_per_pallet(self):
+        """
+        With 1 pallet each for 2 different products.
+        Reserve pick (action_assign).
+        Pick should be split into 2 picks 1 for each pallet, 
+        reusing the original pick.
+        """
+        apple_pallet = self.apple_pallet
+        banana_pallet = self.banana_pallet
+
+        products_info = [{"product": self.apple, "qty": 10}, {"product": self.banana, "qty": 10}]
+        moves = self.create_move(self.picking, products_info)
+
+        apple_move = moves.filtered(lambda m: m.product_id == self.apple)
+        banana_move = moves - apple_move
+
+        self.picking.action_assign()
+
+        apple_pick = apple_move.picking_id
+        banana_pick = banana_move.picking_id
+
+        self._assert_picking_fields(apple_pick, state="assigned", group_name=apple_pallet.name)
+        self._assert_picking_fields(banana_pick, state="assigned", group_name=banana_pallet.name)
+
+        # Check we haven't mangled the moves or move_lines
+        apple_move = apple_pick.move_lines
+        self._assert_move_fields(apple_move, self.apple, 10)
+
+        apple_move_line = apple_pick.move_line_ids
+        self._assert_move_line_fields(apple_move_line, self.apple, 10, apple_pallet)
+
+        banana_move = banana_pick.move_lines
+        self._assert_move_fields(banana_move, self.banana, 10)
+
+        banana_move_line = banana_pick.move_line_ids
+        self._assert_move_line_fields(banana_move_line, self.banana, 10, banana_pallet)
+
+        # Check that the original pick has been reused
+        self.assertTrue(self.picking.exists())
+
+    def test_reserve_2_pallets_same_prods_splits_1_pick_per_pallet(self):
+        """
+        With 2 pallets that contains the same product.
+        Reserve pick (action_assign).
+        Pick should be split into 2 picks 1 for each pallet, 
+        reusing the original pick.
+        """
+        MoveLine = self.env["stock.move.line"]
+
+        apple_pallet = self.apple_pallet
+
+        other_apple_pallet = self.create_package()
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, 10, package_id=other_apple_pallet.id
+        )
+
+        products_info = [{"product": self.apple, "qty": 20}]
+        self.create_move(self.picking, products_info)
+
+        self.picking.action_assign()
+
+        apple_move_lines = MoveLine.search([("product_id.name", "=", "Test product Apple")])
+        self.assertEqual(len(apple_move_lines), 2)
+
+        apple_pallet_move_line = apple_move_lines.filtered(lambda ml: ml.package_id == apple_pallet)
+        other_apple_pallet_move_line = apple_move_lines - apple_pallet_move_line
+
+        self.assertTrue(apple_pallet_move_line.exists())
```

Comment:
We have just search for those move lines, so .exists() is not needed.

Reply:
Have now removed this.